src/lib/support/Span.h

```diff
@@ -38,6 +41,12 @@ class Span
 
     const T * data() const { return mDataBuf; }
     size_t size() const { return mDataLen; }
+    bool empty() const { return mDataBuf == nullptr; }
+    bool data_equal(const Span & other) const
+    {
+        return !empty() && !other.empty() && size() == other.size() &&
+                memcmp(data(), other.data(), size() * sizeof(T)) == 0;
```

Comment:
Suggest:

`return (data() != nullptr) && (other.data() != nullptr) && (size() == other.size()) && (memcmp(data(), other.data(), size() * sizeof(T)) == 0);`

That is closer to the intent and should work irrespective of whether empty() is implemented with nullptr check or size() == 0 check.